server/index.js

```diff
@@ -1,56 +1,89 @@
-const { GraphQLServer, PubSub } = require('graphql-yoga')
-const fetch = require('node-fetch')
-const fs = require('fs')
+const { GraphQLServer, PubSub } = require('graphql-yoga');
+const fetch = require('node-fetch');
+const fs = require('fs');
+const R = require('ramda');
 
-const PATH_TO_GRAPHQL_SCHEMA = '../graphql/schema.graphql'
+const PATH_TO_GRAPHQL_SCHEMA = '../graphql/schema.graphql';
+const UPDATE_FLIGHTS_CHANNEL = 'FLIGHT_UPDATES';
+const REFETCH_DATA_INTERVAL = 3 * 60 * 1000;
 
 const cacheTime = 5 * 60 * 1000;
 
-let cachedData;
 let lastFetchTime;
+let getApiResponsePromise;
 
-async function fetchData() {
+const cachedFlightData = {};
+
+const formatFlightData = flights =>
+  flights.map(flight => ({
+    id: flight.id,
+    orientation: flight.orientation,
+    location: {
+      lat: flight.location.coordinates[0],
+      lng: flight.location.coordinates[1],
+    },
+  }));
+
+const refetchData = async pubsub => {
+  const changedFlights = [];
+  console.warn('Refreshing API data');
+
+  const rawResponse = await fetch(
+    'https://3gyucw6pqd.execute-api.eu-west-1.amazonaws.com/dev'
+  );
+
+  if (!rawResponse.ok) {
+    console.warn('API is down');
+    throw new Error(`API is down`);
+  }
+
+  const data = await rawResponse.json();
+
+  data.flights.forEach(flight => {
+    if (!R.equals(flight, cachedFlightData[flight.id])) {
```

Comment:
This takes care of the flights that have been modified. What about the flights that are no longer available?